apps/io-services-cms-webapp/src/utils/subscription.ts

```diff
@@ -64,8 +86,14 @@ export const serviceOwnerCheckManageTask = (
       )
     ),
     TE.chainW((serviceSubscription) =>
-      serviceSubscription.ownerId === userId
-        ? TE.right(serviceId)
-        : TE.left(ResponseErrorForbiddenNotAuthorized)
+      pipe(
+        serviceSubscription.ownerId,
+        extractOwnerId,
+        TE.chainW((ownerId) =>
```

Comment:
maybe we can use `TE.filterOrElse`?

Reply:
We cannot use TE.filterOrElse, cause we want to return the serviceId on the right and not the ownerId, so we need the chainW

``` typescript
     pipe(
        serviceSubscription.ownerId,
        extractOwnerId,
        TE.fromEither,

        TE.filterOrElseW(
          (ownerId) => ownerId === userId, // this will return TE.right(ownerId) which is wrong
          () => ResponseErrorForbiddenNotAuthorized 
        ),
```